ext/random/php_random_csprng.h

```diff
@@ -20,7 +20,9 @@
 
 # include "php.h"
 
-PHPAPI zend_result php_random_bytes(void *bytes, size_t size, bool should_throw);
+ZEND_ATTRIBUTE_NONNULL PHPAPI zend_result php_random_bytes(void *bytes, size_t size, bool should_throw);
+ZEND_ATTRIBUTE_NONNULL PHPAPI zend_result php_random_bytes_ex(void *bytes, size_t size, char *errstr, size_t errstr_size);
```

Comment:
I prefer Clang's method:

```c
#ifndef _Nonnull
#define _Nonnull
#endif

#ifndef _Nullable
#define _Nullable
#endif

PHPAPI zend_result php_random_bytes(
    void * _Nonnull bytes,
    size_t size,
    bool should_throw
);

PHPAPI zend_result php_random_bytes_ex(
    void * _Nonnull bytes,
    size_t size,
    char * _Nullable errstr,
    size_t errstr_size
);
```
Regardless of the attributes, I would not want to bother with an error string sometimes. Is there any reason you'd want to force all callers to populate this error string?

Reply:
> I prefer Clang's method

Placing the attribute next to the parameters themselves is nice indeed, but I'm afraid GCC doesn't support this syntax. This would make the attribute less useful. `__attribute__((nonnull))` has the benefit of being compatible with at least GCC and Clang.

> Regardless of the attributes, I would not want to bother with an error string sometimes. Is there any reason you'd want to force all callers to populate this error string?

There are multiple variants of this API, and one of them allows us to ignore errors entirely: https://github.com/php/php-src/blob/93c20de2f05bf5b5984fa04736756ae87c2e5a12/ext/random/php_random_csprng.h#L31

BTW, did you have a chance to look at https://github.com/php/php-src/pull/14242 ?